Packs/SplunkPy/Integrations/SplunkPy/SplunkPy.py

```diff
@@ -31,6 +31,360 @@
+            self.id = data[EVENT_ID]
+
+    @property
+    def id(self):
+        return self.id
+
+    @id.setter
+    def id(self, notable_id):
+        self.id = notable_id
+
+    @property
+    def data(self):
+        return self.data
+
+    @data.setter
+    def data(self, data):
+        self.data = data
+
+    @property
+    def status(self):
+        return self.status
+
+    @status.setter
+    def status(self, status):
+        self.status = status
+
+    @classmethod
+    def from_json(cls, notable_dict):
+        """ Deserialization method.
+
+        Args:
+            notable_dict: The notable dict in JSON format.
+
+        Returns:
+            An instance of the Notable class constructed from JSON representation.
+
+        """
+        return cls(
+            data=notable_dict.get(DATA),
+            notable_id=notable_dict.get(DATA).get(EVENT_ID),
+            status=notable_dict.get(STATUS)
+        )
+
+
+class EnrichmentStatus(object):
+    """ Contains all enrichment's possible status during the enrichment process """
+    NOT_YET_ENRICHED = 'Notable not yet enriched'
+    DONE_ENRICHING = 'Enrichment has been submitted to Splunk'
+    DONE_HANDLING = 'Enrichment has been handled successfully'
+    CREATED_INCIDENT_ENRICH_FAILURE = 'Created an incident from the raw notable as enrichment failed'
+    EXCEEDED_TIMEOUT = 'Enrichment has exceeded timeout'
+    HANDLED_STATUS = (DONE_HANDLING, EXCEEDED_TIMEOUT)
+
+
+class Enrichment:
+    """ A class to represent an enrichment.
+
+    Args:
+        notable (Notable): The notable to be enriched.
+        Possible values are: Drilldown, Asset & Identity.
+        jobs (list[Job]): The list of jobs.
+        enrichment_id (str): The enrichment ID.
+        status (str): The enrichment's status.
+
+    Attributes:
+        notable (Notable): The notable to be enriched.
+        id (str): The enrichment's id (notable's id in Splunk server).
+        status (str): The enrichment's status.
+        jobs (list): The list of all jobs that the enrichment needs to handle.
+
+    """
+    def __init__(self, notable, jobs=None, enrichment_id=None, status=None):
+        self.notable = notable
+        self.id = enrichment_id if enrichment_id else notable.id
+        self.status = status if status else EnrichmentStatus.NOT_YET_ENRICHED
+        if jobs:
+            self.jobs = jobs
+        else:
+            self.jobs = []
+            if DRILLDOWN_ENRICHMENT in ENABLED_ENRICHMENTS:
+                self.drilldown_job = Job(job_type=DRILLDOWN_ENRICHMENT)
+                self.jobs.append(self.drilldown_job)
+            if ASSET_ENRICHMENT in ENABLED_ENRICHMENTS:
+                self.asset_job = Job(job_type=ASSET_ENRICHMENT)
+                self.jobs.append(self.asset_job)
+            if IDENTITY_ENRICHMENT in ENABLED_ENRICHMENTS:
+                self.identity_job = Job(job_type=IDENTITY_ENRICHMENT)
+                self.jobs.append(self.identity_job)
```

Comment:
No need to create job objects on the init, only when we initiate an enrichemnt.
```suggestion
    def __init__(self, notable, jobs=[], enrichment_id=None, status=None):
        self.notable = notable
        self.id = enrichment_id if enrichment_id else notable.id
        self.status = status if status else EnrichmentStatus.NOT_YET_ENRICHED
        self.jobs = jobs
```